test/System.Web.Http.Cors.Test/CorsMessageHandlerTest.cs

```diff
@@ -180,6 +189,40 @@ public async Task SendAsync_HandlesExceptions_ThrownDuringPreflight()
             Assert.Equal(HttpStatusCode.MethodNotAllowed, response.StatusCode);
         }
 
+        [Fact]
+        public async Task SendAsync_Preflight_RethrowsExceptions_WhenRethrowFlagIsTrue()
+        {
+            HttpConfiguration config = new HttpConfiguration();
+            config.Routes.MapHttpRoute("default", "{controller}");
+            HttpServer server = new HttpServer(config);
+            CorsMessageHandler corsHandler = new CorsMessageHandler(config, true);
+            corsHandler.InnerHandler = server;
+            HttpMessageInvoker invoker = new HttpMessageInvoker(corsHandler);
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Options, "http://localhost/sample");
+            request.SetConfiguration(config);
+            request.Headers.Add(CorsConstants.Origin, "http://localhost");
+            request.Headers.Add(CorsConstants.AccessControlRequestMethod, "RandomMethod");
+
+            await Assert.ThrowsAsync<HttpResponseException>(() => invoker.SendAsync(request, CancellationToken.None));
+        }
+
+        [Fact]
+        public async Task SendAsync_RethrowsExceptions_WhenRethrowFlagIsTrue()
+        {
+            HttpConfiguration config = new HttpConfiguration();
+            config.Routes.MapHttpRoute("default", "{controller}");
+            config.Services.Replace(typeof(IExceptionHandler), new PassthroughExceptionHandler());
```

Comment:
This is needed to reproduce the exact scenario of original issue, otherwise default implementation of IExceptionHandler maps Exceptions to just 500 status code